api-sdk/src/main/java/com/smartling/api/sdk/exceptions/ApiException.java

```diff
@@ -23,13 +34,68 @@
 {
     private static final long serialVersionUID = -397098626101615761L;
 
-    public ApiException(final String message)
+    private int httpCode;
+    private List<String> messages = new ArrayList<>();
+
+    ApiException(final String contents, List<String> messages, int httpCode)
     {
-        super(message);
+        super(contents);
+        this.messages = messages;
+        this.httpCode = httpCode;
     }
 
-    public ApiException(final Exception e)
+    ApiException(final Exception e)
     {
         super(e);
+        messages.add(e.getMessage());
+    }
+
+    public static ApiException newException(String contents, int httpCode)
+    {
+        ApiResponse<EmptyResponse> apiResponse = JsonReader.parseApiResponse(contents, new TypeToken<ApiResponseWrapper<EmptyResponse>>()
+                {
+                }
+        );
+        String apiCode = apiResponse.getCode();
+        List<String> messages = apiResponse.getMessages();
+        return getApiException(contents, httpCode, apiCode, messages);
```

Comment:
With current solution I can pass it, but we have to discuss your and @PavloMyrotiuk comments regarding this class